src/components/ActivityForm.js

```diff
@@ -49,13 +50,30 @@ export default class ActivityForm extends React.Component {
 
   onDayChange = e => this.setState({ day: e.target.value });
 
+  onBlur = e =>
+    !e.target.value && this.setState({ error: `Please provide ${e.target.placeholder}` });
+
   onSubmit = e => {
     e.preventDefault();
     const { name, day, classNo, room, teacher } = this.state;
-    if (!name || !day || !classNo) {
-      this.setState({ error: 'Please provide Activity Name, Day and Class Number!' });
-      // TODO: What if I provide Activity Name and Day, but not Class Number?
-      // For user it will be confusing ;) but this is to handle by separate PR!
+    if (!name || !day || !classNo || !teacher || !room) {
```

Comment:
and then, this logic can be moved to Input. You should have as many errors as required fields, separate ones.